src/main/java/picocli/CommandLine.java

```diff
@@ -17840,7 +17840,18 @@ static final boolean forceEnabled() { return System.getenv("CLICOLOR_FORCE") !=
                     && !"0".equals(System.getenv("CLICOLOR_FORCE"));}
             /** http://stackoverflow.com/questions/1403772/how-can-i-check-if-a-java-programs-input-output-streams-are-connected-to-a-term */
             static boolean calcTTY() {
-                try { return System.class.getDeclaredMethod("console").invoke(null) != null; }
+                try {
+                    Object console = System.class.getDeclaredMethod("console").invoke(null);
+                    if (console == null) {
+                        return false;
+                    }
+                    try {
+                        Method isTerminal = Console.class.getDeclaredMethod("isTerminal");
```

Comment:
Looks good, but the `Console` class was introduced in Java 6, and picocli supports Java 5, so can you change this to find the `Console` class via reflection?

Reply:
Thanks, done